Microsoft.Toolkit.Uwp.UI.Controls/Menu/Menu.Events.cs

```diff
@@ -0,0 +1,244 @@
+        private const string ShiftValue = "SHIFT";
+
+        private static bool NavigateThroughMenuHeader(KeyEventArgs args, Menu menu, Orientation orientation)
+        {
+            if (orientation == Orientation.Horizontal)
+            {
+                if (args.VirtualKey == VirtualKey.Down)
+                {
+                    menu.SelectedHeaderItem.ShowMenu();
+                    return true;
+                }
+
+                if (args.VirtualKey == VirtualKey.Left)
+                {
+                    MoveFocusBackwardAndGetPrevious(menu);
+                    return true;
+                }
+
+                if (args.VirtualKey == VirtualKey.Right)
+                {
+                    MoveFocusForwardAndGetNext(menu);
+                    return true;
+                }
+            }
+            else
+            {
+                if (args.VirtualKey == VirtualKey.Right)
+                {
+                    menu.SelectedHeaderItem.ShowMenu();
+                    return true;
+                }
+
+                if (args.VirtualKey == VirtualKey.Up)
+                {
+                    MoveFocusBackwardAndGetPrevious(menu);
+                    return true;
+                }
+
+                if (args.VirtualKey == VirtualKey.Down)
+                {
+                    MoveFocusForwardAndGetNext(menu);
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        private static bool NavigateThroughMenuFlyoutItems(KeyEventArgs args, Menu menu, Orientation orientation)
+        {
+            if (orientation == Orientation.Horizontal)
+            {
+                if (args.VirtualKey == VirtualKey.Left)
+                {
+                    menu.SelectedHeaderItem.HideMenu();
+                    MoveFocusBackwardAndGetPrevious(menu).ShowMenu();
+                    return true;
+                }
+
+                if (args.VirtualKey == VirtualKey.Right)
+                {
+                    menu.SelectedHeaderItem.HideMenu();
+                    MoveFocusForwardAndGetNext(menu).ShowMenu();
+                    return true;
+                }
+            }
+            else
+            {
+                if (args.VirtualKey == VirtualKey.Up)
+                {
+                    menu.SelectedHeaderItem.HideMenu();
+                    MoveFocusBackwardAndGetPrevious(menu).ShowMenu();
+                    return true;
+                }
+
+                if (args.VirtualKey == VirtualKey.Down)
+                {
+                    menu.SelectedHeaderItem.HideMenu();
+                    MoveFocusForwardAndGetNext(menu).ShowMenu();
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        // this function enables the menu to cycle from start to end and from end to start
+        private static MenuItem MoveFocusForwardAndGetNext(Menu menu)
```

Comment:
it is going to be more complex for any other contributor to understand what it does :)